projects/packages/videopress/src/class-access-control.php

```diff
@@ -0,0 +1,367 @@
+	}
+
+	/**
+	 * Get the singleton instance.
+	 *
+	 * @return AJAX
+	 */
+	public static function instance() {
+		if ( self::$instance === null ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Determines if Jetpack Memberships are available.
+	 *
+	 * @return bool
+	 */
+	private function jetpack_memberships_available() {
+		return class_exists( '\Jetpack_Memberships' );
+	}
+
+	/**
+	 * Determines if Jetpack Subscriptions are available.
+	 *
+	 * @return bool
+	 */
+	private function jetpack_subscriptions_available() {
+		if ( function_exists( '\Automattic\Jetpack\Extensions\Premium_Content\subscription_service' ) ) {
+			return true;
+		}
+
+		if ( ! defined( 'JETPACK__PLUGIN_DIR' ) ) {
+			return false;
+		}
+
+		$subscription_service_file_path = JETPACK__PLUGIN_DIR . 'extensions/blocks/premium-content/_inc/subscription-service/include.php';
+		if ( ! file_exists( $subscription_service_file_path ) ) {
+			return false;
+		}
+
+		require_once $subscription_service_file_path;
+
+		return function_exists( '\Automattic\Jetpack\Extensions\Premium_Content\subscription_service' );
+	}
+
+	/**
+	 * Check default user access. By default, subscribers or higher can view videos.
+	 *
+	 * @param WP_Post $post_to_check The post to check.
+	 *
+	 * @return bool
+	 **/
+	private function get_default_user_capability_for_post( $post_to_check ) {
+		global $post;
+		$old_post = $post;
+
+		// phpcs:disable WordPress.WP.GlobalVariablesOverride.Prohibited
+		$post         = $post_to_check;
+		$default_auth = current_user_can( 'read' );
+
+		// phpcs:disable WordPress.WP.GlobalVariablesOverride.Prohibited
+		$post = $old_post;
+
+		return $default_auth;
+	}
+
+	/**
+	 * Determines if the current user can access restricted content and builds the restriction_details array.
+	 *
+	 * @param string $guid the video guid.
+	 * @param int    $embedded_post_id the post id.
+	 * @param int    $selected_plan_id the selected plan id if applicable.
+	 *
+	 * @return array
+	 */
+	private function build_restriction_details( $guid, $embedded_post_id, $selected_plan_id ) {
+		$post_to_check = get_post( $embedded_post_id );
+
+		if ( empty( $post_to_check ) ) {
+			$restriction_details = $this->default_video_restriction_details( false );
+			return $this->filter_video_restriction_details( $restriction_details, $guid, $embedded_post_id, $selected_plan_id );
+		}
+
+		$default_auth        = $this->get_default_user_capability_for_post( $post_to_check );
+		$restriction_details = $this->default_video_restriction_details( $default_auth );
+
+		if ( $this->jetpack_memberships_available() ) {
+			$memberships_can_view_post         = \Jetpack_Memberships::user_can_view_post( $embedded_post_id );
+			$restriction_details               = $this->get_subscriber_only_restriction_details( $default_auth );
+			$restriction_details['can_access'] = $memberships_can_view_post;
+			return $this->check_block_level_access(
```

Comment:
This `return` could be removed since the same call is made outside the if statement with the same vars.